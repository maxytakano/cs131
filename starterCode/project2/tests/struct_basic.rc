structdef MYSTRUCT{
	int id;
	int i[10];
	float f[10];
	MYSTRUCT* next;
};

MYSTRUCT mystruct_g;

function : void main(){
	MYSTRUCT  myarr[10];
	MYSTRUCT * myptr = &myarr[0];
	int i = 0;
	while(i < 10){
		myarr[i].id = i;
		myarr[7].i[i] = i;
		myarr[7].f[i] = i * 10.7;
		myarr[i].next = &(myptr[i+1]);
		i++;
	}
	myarr[9].next = *myarr;
	mystruct_g = myarr[7];
	myptr = &mystruct_g;
	i = 0;
	while(i < 10){
		cout<<mystruct_g.i[i]<<"\t"<<mystruct_g.f[i]<<endl;
		i++;
	}
	i = 0;
	while(i < 10){
		cout<<myptr->id<<endl;
		myptr = myptr->next;
		i++;
	}
	i = 0;
	while(i < 10){
		cout<<myptr->i[i]<<"\t"<<myptr->f[i]<<endl;
		i++;
	}
}
