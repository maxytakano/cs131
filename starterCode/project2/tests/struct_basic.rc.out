0	0.00
1	10.70
2	21.40
3	32.10
4	42.80
5	53.50
6	64.20
7	74.90
8	85.60
9	96.30
7
8
9
10
Attempt to dereference NULL pointer.
